src/Unity.Mathematics/quaternion.cs

```diff
@@ -345,6 +345,277 @@ public static quaternion Euler(float x, float y, float z, RotationOrder order =
             return Euler(float3(x, y, z), order);
         }
 
+        /// <summary>
+        /// Returns the Euler angle representation of the quaternion following the XYZ rotation order.
+        /// All rotation angles are in radians and clockwise when looking along the rotation axis towards the origin.
+        /// </summary>
+        /// <param name="q">The quaternion to convert to Euler angles.</param>
+        /// <returns>The Euler angle representation of the quaternion in XYZ order.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float3 EulerXYZ(quaternion q)
```

Comment:
quaternion conversion to euler methods currently live in the quaternion struct which gives us a bit of a weird call:
`float3 eulerAngles = quaternion.EulerXYZ(myquat);`
I put them here initially because the others were defined in the quaternion struct (functions to convert euler to quat).
Maybe they should live in float3 ?
`float3 eulerAngles = float3.EulerXYZ(myQuat);`
Or otherwise the math class altogether?
`float3 eulerAngles = math.EulerXYZ(myQuat);`

@mecanimdev @unpacklo thoughts?

Reply:
> Maybe they should live in float3 ?
> `float3 eulerAngles = float3.EulerXYZ(myQuat);`

I think this makes the most sense. We would need to add a new `float3.cs` file to hold these manually written methods instead of stuffing them into `float3.gen.cs`. @oldio Did you want me to move that around for you?

Reply:
Feel free to do edits on this PR @unpacklo if you have the bandwidth. I don't think I'll have much time to get to it today unfortunately.